docs/check-examples

```diff
@@ -118,7 +118,7 @@ do
   ## For header lines, extract the error message from
   ## the header text (everything but the initial ##[ ]* )
   ( echo "$line" | grep -E -e '[#][#] ' > /dev/null ) \
-   && msg=`echo "$line" | sed -E -e 's/^##[ ]*[*]*//' -e 's/[*]*$//' -e 's/\\*/\\\\*/' -e 's/_[a-z]+_/.*/g'`
+   && msg=`echo "$line" | sed -E -e 's/ {#.*}//' | sed -E -e 's/^##[ ]*[*]*//' -e 's/[*]*$//' -e 's/\\*/\\\\*/' -e 's/_[a-z]+_/.*/g'`
```

Comment:
Can you explain me what this change does? At least I would love if there was a change in the comment that reflects what the change on this obscure regex is ^^

Reply:
I added a comment in the script. It converts the error message in the header into a regex to match against the eror message produced by running dafny against the example, to be sure the example does what I say.